MiniAn/minian_main.py

```diff
@@ -0,0 +1,366 @@
+        if defs.DoricFile.Attribute.OPERATIONS not in params_source_data:
+            params_source_data[defs.DoricFile.Attribute.OPERATIONS] = params_source_data[defs.DoricFile.Attribute.OPERATION_NAME]
+
+        del params_source_data[defs.DoricFile.Attribute.OPERATION_NAME]
+
+    if minian_parameters.parameters[defs.Parameters.danse.SPATIAL_DOWNSAMPLE] > 1:
+        minian_parameters.parameters[defs.DoricFile.Attribute.BINNING_FACTOR] = minian_parameters.parameters[defs.Parameters.danse.SPATIAL_DOWNSAMPLE]
+
+    mn_utils.save_minian_to_doric(
+        Y, A, C, AC, S,
+        fr = minian_parameters.fr,
+        bits_count = attrs[defs.DoricFile.Attribute.BIT_COUNT],
+        qt_format = attrs[defs.DoricFile.Attribute.FORMAT],
+        username = attrs.get(defs.DoricFile.Attribute.USERNAME, sensor),
+        vname = minian_parameters.params_load_doric['fname'],
+        vpath = f"{defs.DoricFile.Group.DATA_PROCESSED}/{driver}/",
+        vdataset = f"{series}/{sensor}/",
+        params_doric = minian_parameters.parameters,
+        params_source = params_source_data,
+        saveimages = True,
+        saveresiduals = True,
+        savespikes = True
+    )
+
+    # Close cluster
+    client.close()
+    cluster.close()
+
+
+
+def preview(minian_parameters):
+    """minian_preview
+    """
+
+    # Start cluster
+    print(mn_defs.Messages.Main.START_CLUSTER, flush=True)
+    cluster = LocalCluster(**minian_parameters.params_LocalCluster)
+    annt_plugin = TaskAnnotation()
+    cluster.scheduler.add_plugin(annt_plugin)
+    client = Client(cluster)
+
+    # MiniAn CNMF
+    intpath = os.path.join(minian_parameters.paths[defs.Parameters.Path.TMP_DIR], mn_defs.Text.Main.INTERMEDIATE)
+    subset = {"frame": slice(*minian_parameters.preview_parameters[defs.Parameters.Preview.RANGE])}
+
+    file_, chk, varr_ref = load_chunk(intpath, subset, minian_parameters)
+
+    varr_ref = preprocess(varr_ref, intpath, minian_parameters)
+
+    Y, Y_fm_chk, Y_hw_chk = correct_motion(varr_ref, intpath, chk, minian_parameters)
+
+    seeds_final, max_proj = initialize_seeds(Y_fm_chk, Y_hw_chk, minian_parameters)
+
+    # Save data for preview to hdf5 file
+    try:
+        with h5py.File(minian_parameters.preview_parameters[defs.Parameters.Preview.FILEPATH], 'w') as hdf5_file:
+
+            if minian_parameters.preview_parameters[mn_defs.Preview.MAX_PROJ_DATASET_NAME] in hdf5_file:
+                del hdf5_file[minian_parameters.preview_parameters[mn_defs.Preview.MAX_PROJ_DATASET_NAME]]
+
+            hdf5_file.create_dataset(minian_parameters.preview_parameters[mn_defs.Preview.MAX_PROJ_DATASET_NAME], data = max_proj.values, dtype='float', chunks = True)
+
+            groupseed = hdf5_file.create_group(minian_parameters.preview_parameters[mn_defs.Preview.SEED_GROUP_NAME])
+            for key in seeds_final:
+                groupseed.create_dataset(key, data = seeds_final[key], dtype = 'float',chunks = True)
```

Comment:
I think this way, we're saving multiple times the same coordinates. I propose to save seeds like that:
```
with h5py.File(filename, 'a') as f:
    f.create_dataset('Seeds', data = seeds[['height', 'width']], dtype='int', chunks = True)
    f['Seeds'].attrs['Merged'] = seeds.index[seeds['mask_mrg'] == True].tolist()
    f['Seeds'].attrs['Refined']  = seeds.index[(seeds['mask_ks'] == True) & (seeds['mask_pnr'] == True)].tolist()
```

Reply:
I don't think it is saving multiple time the same coodinates. Each dataset created is a column of the seeds datasetframe. So one dataset for the `height` one dataset `width` one for the `mask_mrg` and so on.

I was just saving the raw data. I didn't think farther how to say it. Saving the coordinates in a 2D matrix and them using the attribute for the Merging or not is maybe smarter